forem/forem
# Fix text color contrast on social previews

## What type of PR is this? (check all applicable)


## Description
- The text color that appears on social previews depends on the author's brand color. However, when a light color is chosen, the text may not have enough contrast to meet WCAG standards.
- This PR addresses the issue by using black texts if the contrast is too low.

## Related Tickets & Documents



- Closes #18772 

## QA Instructions, Screenshots, Recordings

1. Log in
2. Go to ```/settings```
3. Scroll down to Branding section. Change the brand color to #f0f4f5.
4. Create a post and take note of the article_id.
5. Go to ```/social_previews/article/:article_id.html``` 
6. Check that the text to background contrast ratio is at least 4.5:1

Screenshots:

![brand color](https://user-images.githubusercontent.com/88938117/220275063-99cb9c32-9fa6-4e42-811d-4a339fdcffcd.png)
![contrast ratio](https://user-images.githubusercontent.com/88938117/220275106-bc258b7e-eaea-4616-b3be-093531fb1ef9.jpg)

## Added/updated tests?

- [x] No, and this is why: Used a class that has tests written previously

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?
